python_modules/dagster/dagster_tests/definitions_tests/asset_reconciliation_tests/definition_change_scenarios.py

```diff
@@ -0,0 +1,45 @@
+"""Scenarios where the set of asset definitions changes between ticks.
+"""
+
+from .asset_reconciliation_scenario import AssetReconciliationScenario, run_request
+from .basic_scenarios import one_asset, two_assets_in_sequence
+from .partition_scenarios import one_asset_one_partition
+
+definition_change_scenarios = {
+    "asset_removed": AssetReconciliationScenario(
+        assets=[],
+        unevaluated_runs=[],
+        expected_run_requests=[],
+        cursor_from=AssetReconciliationScenario(
+            assets=one_asset,
+            unevaluated_runs=[],
+        ),
+    ),
+    "downstream_asset_removed": AssetReconciliationScenario(
+        assets=one_asset,
+        unevaluated_runs=[],
+        expected_run_requests=[],
+        cursor_from=AssetReconciliationScenario(
+            assets=two_assets_in_sequence,
+            unevaluated_runs=[],
+        ),
+    ),
+    "partitioned_asset_removed": AssetReconciliationScenario(
+        assets=[],
+        unevaluated_runs=[],
+        expected_run_requests=[],
+        cursor_from=AssetReconciliationScenario(
+            assets=one_asset_one_partition,
+            unevaluated_runs=[],
+        ),
+    ),
+    "partitions_def_removed": AssetReconciliationScenario(
+        assets=one_asset,
+        unevaluated_runs=[],
+        expected_run_requests=[run_request(asset_keys=["asset1"])],
```

Comment:
what happens when a partitions def is added to a previously unpartitioned asset that was materialized by the reconciliation sensor? do we kick off partitioned runs?

Reply:
yes - I just added a test for this